seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java

```diff
@@ -627,7 +629,11 @@ private void updateMetricsContextInImap() {
                     boolean unLockedIMap = false;
                     while (!unLockedIMap) {
                         try {
-                            metricsImap.unlock(Constant.IMAP_RUNNING_JOB_METRICS_KEY);
+                            if (System.currentTimeMillis() - lockTs > 5000) {
+                                logger.severe("unlock imap: " + Constant.IMAP_RUNNING_JOB_METRICS_KEY + " exceeds 5s, exit while loop.");
+                            } else {
+                                metricsImap.unlock(Constant.IMAP_RUNNING_JOB_METRICS_KEY);
```

Comment:
```suggestion
                                metricsImap.forceUnlock(Constant.IMAP_RUNNING_JOB_METRICS_KEY);
```
or change to forceUnlock would be a good choice.

Reply:
Sorry, I am not very familiar with the Hazelcast framework and I am not sure about the difference between forceUnlock and unlock. If I change to forceUnlock, do I still need a while loop?